### DISCUSSION TITLE
No GPS fix with 3.9.10 on LineageOS 17.1 with Google Play services

### DISCUSSION THREAD

**Discussion by zonque** [osmandapp/OsmAnd#11408] [Q&A] [answered]
### Description

On a newly installed LineageOS 17.1 (device tissot) OsmAnd does not get a GPS fix though many other apps do. This has also been reported in #10637 (which is locked unfortunately).

Installing the [nightly](https://download.osmand.net/latest-night-build/OsmAnd-default.apk) resolves the issue, so whatever has been done in that area did fix a bug. I'm only reporting this to make sure the fix is recognized as such, as there wasn't any conclusion in #10637. I'll patiently wait for the next official release to be published.

### How to reproduce?

Open OsmAnd. The location icon in the upper right corner is not showing up, and even though the "GPS Status & Toolbox" app sees satellites and has a fix, OsmAnd doesn't.

![image](https://user-images.githubusercontent.com/106304/114272443-80745e00-9a16-11eb-9fd8-e64d90ed3ba3.png)

### Your Environment
OsmAnd Version: 3.9.10
Android/iOS version: LineageOS 17.1-20210406 (Android 10)
Device model: Xiaomi A1 (tissot)

**Comment by zonque** [comment]
After some more investigation, the weird thing is that OsmAnd nightly seems to be the only app on my device besides "GPS Status & Toolbox" that is able to detect the GPS based location correctly. Could anyone summarize what has changed in OsmAnd between 3.9.10 and nightly?

**Comment by scaidermern** [comment]
Did you try the OsmAnd version from F-Droid?

**Comment by zonque** [comment]
> Did you try the OsmAnd version from F-Droid?

No, only Google Play. I'm also a Plus subscriber.

**Comment by scaidermern** [comment]
Sorry, I misread that you actually _have_ Google Play Services. This is strange.

I also use OsmAnd+ on LineageOS 17.1 with Android 10 and with Google Play services. GPS works fine. However my LineageOS 17.1 is a little bit older (I should update to the latest version I guess... or maybe not?).

Anyway, you can still try the F-Droid version just to find out if it has the same problem.

**Comment by zonque** [comment]
Yeah, but what puzzles me is that the nightly works. There must be a reason for that I suppose? Other apps are also affected by this, and my hopes are that by understanding what fixed OsmAnd, I can find a way to fix other things as well.

**Comment by vshcherb** [comment, marked as answer]
Essential difference between Nightly and Google Play that Nightly uses native Android GPS API and  Google Play version uses Google Play Services Location. 

Looks like Google Play Services Location API doesn't work correctly on your devices. Note that other apps could be old and still using Android GPS API like "GPS Status & Toolbox"

**Reply by zonque** [reply]
Okay, thanks a lot for that summary. That makes sense. Will dig in that direction then! :pray:

**Reply by vshcherb** [reply]
There could be some limitation for Google Play Service Location anyway if you want to use without it, you are free to use the nightly version.

**Comment by zonque** [comment]
For reference, the issue was resolved as described here: https://gitlab.com/LineageOS/issues/android/-/issues/3306

**Reply by BenoMueller** [reply]
I use LineageOS 18.1 (and Gapps) and every other app (e.g. LocusMap) gets a GPS fix without any problems- except for OsmAnd. And indeed, granting permissions as described under the link above solves the problem for OsmAnd. Yet, after having granted the permissions, every time I turn on my mobile phone, Google Play Services access my location... So I "revoked" again the permissions and instead installed the nightly version.

**Reply by VelAstra** [reply]
I use Android 11 Arrow OS for Max Pro M1 and my GPS doesnt work, that link worked, now my GPS works perfectly, thanks

**Comment by editor-hu** [comment]
Same problem here. I use Huawei P10 phone and Android 7.0
Used OSMAND successfully for long time. Yesterday tried to use again, no actual position detected. Osmand app has the permission to have location. Android location is set to high accuracy, GPS + Wifi + stations. It worked and now it does not and I can't find why. Any helpful suggestion would be much appreciated.
